torch/csrc/jit/tracer.cpp

```diff
@@ -134,33 +149,12 @@ Value* getValueTrace(const IValue& var) {
     oss << "Tried to trace Future that the tracer was not aware of.";
     throw std::runtime_error(oss.str());
   } else {
-    std::ostringstream oss;
-    oss << "Unknown type used in value trace lookup!";
-    throw std::runtime_error(oss.str());
-  }
-}
-
-// allow tracing of tuples passed to List[Tensor] or Tuple[Tensor...] arguments
-// One might merge getValueTrace and getNestedValueTrace after checking that
-// casting to IValue instead  of Variable is OK
-Value* getNestedValueTrace(const IValue& v) {
-  auto& state = getTracingState();
-  if (v.isTensorList()) {
-    return state->graph
-        ->insertNode(state->graph->createList(
-            TensorType::get(),
-            fmap(
-                v.toTensorListRef(),
-                [](const IValue& val) { return getNestedValueTrace(val); })))
-        ->output();
-  } else if (v.isTuple()) {
-    return state->graph
-        ->insertNode(state->graph->createTuple(fmap(
-            v.toTuple()->elements(),
-            [](const IValue& val) { return getNestedValueTrace(val); })))
-        ->output();
+    // If the values are non-tensors, we try to create constants 
```

Comment:
insertConstant of arbitrary IValues can fail with an internal assert. This needs to only be called if we _can_ create a constant from var.

Reply:
I checked that `graph->insertConstant` already call `tryInsertConstant` and do the error handling after https://github.com/pytorch/pytorch/pull/19686, it will throw constant_not_support error if the ivalues are not valid. Shall I simply add a test for it and we are good to go?